### DISCUSSION TITLE
AOC 2022 Day One

### DISCUSSION THREAD

**Discussion by tkshill** [Virtual-Coffee/virtualcoffee.io#755] [Advent of Code 2022]
Discussion of solutions for [Advent of Code Day 1](https://adventofcode.com/2022/day/1) here

**Comment by bogdaaamn** [comment]
It's wayyy to early to be efficient or creative 😭 Reducers will do today https://github.com/BogDAAAMN/advent-of-code-2022/blob/main/day-01.js

**Reply by danieltott** [reply]
Gah totally forgot about `Math.max` - nice!

**Reply by tkshill** [reply]
Because dotnet has `ReadAllLines`, I didn't consider splitting the string on the newlines. It's about the same amount of code, but I wouldn't have had to do an explicit fold.  This is neat!

**Reply by clandau** [reply]
much nicer than mine, splitting the input twice was 🔑 and would have saved me some lines

**Comment by danieltott** [comment]
Reducers ftw! [Here's mine](https://www.typescriptlang.org/play?pretty=true#code/IYZwngdgxgBAZgV2gFwJYHsIwOYFNkAiwywAFHKgDa4TAC2uAXCMgE6oTYCUjACq+jqoQuADwt2nAHwwA3gFgAUAEgomFjFa4QABxgBeGMADuwVMnj4oAC1IBya8mQ6QjAPRvgO1ADps56wQAIx81Ojd-FhA3AFYARig4YAB2ADYAE3TgAE5cYABmOAAmbLiAFjK4gA5gdIAGIOAq1PyYursAGjklZWVrPPTcVldulV6AQSgoXB1kRhg7Lx1KVChiDAg3ADcIdL8A4J9WEzse5QBfHvOuJR61CA1Ii0MTMwstXR8AKxBMUhvFD0tMgEKwsE8fBRqCAANpQmj0XAAXVCmGQNGQSkugMUoEgsEQKA2MCyYAA8hBcP9Rso7uoLFkSAYjKZzDh8EQSPZSZhcD5kAAPZB2AF0h4WACilDgAGFgJR0OxcAAZYTPEnEYBnDwwXQrCwSDjYGAcZDoIxYYCsY5gGDoOAwXCULbaM4+PXmUgAIgAOhA-V6Ab0dXBFY7gDZHdKuvcXawDWwjXaHWsFUqQDAzUZrcAwG66F5yOh0OkM-oZKGSyAzsp3ctPb6IIGa0dcOkENNSKQzSRKDH5VwDDIe-KYABqGA6K0iACSEGQpFTXC6dSDylFKh1vCtFnGMAAtPAOOlM-0YJQrXgNCPKGKNHR0Cw5WnUNpmVLZfLFa-VSxW+3OzORdQS0edlUvbRkC6G9B3LTN0F7GAZCgECMXA1grwsAB+eDEPmFDrTQiCWC6M46h6DdVHUdBqB8BVsG9B8NFTb9tEYL0uiY5Bn1YkBKK3HcYAAIQPI9dlPXBdQQOhkwk+C9GQawtFwO8LDNHQABUlNwSTDA-HilV-TExh1X54zklYWDdMyF1IYAuiCWCZAUMZlGBUEsCCUTgAAbjOa5tTcHBji8xTJPU09lOslZOzqDp8jXHVahPMKZLNPAwtYN0tAAqk7IcpyjHHGAghXSj7l+Wj6O9CLFOU9joPQTTtNwDdsSUUkKSpLggA)

**Reply by danieltott** [reply]
[Here's an alternate](https://www.typescriptlang.org/play?pretty=true#code/IYZwngdgxgBAZgV2gFwJYHsIwOYFNkAiwywAFHKgDa4TAC2uAXDCMgE6oTYCUzACm3R1UIXAB5WHLgD4YAbwCwAKACQUTKxhtcIAA4wAvDGAB3YKmTx8UABallKlQHIbyZLpCMA9F+C7UAHTYFjYIAEYB6nRewawgXgCsAIxQcMAA7ABsACbZwACcuMAAzHAATPlJACxVSQAcwNkADGHAdZnFCU1OADQOKoqqjjZF2bhsnvL9jgCCUFC4usjMTn66lKhQxBgQXgBuENlBIeEBbKa90wC+fUM3DtwA3MoO6hCasZZGpuaW2noBABWIEwpCeL1U2mQCDYWE+AQo1BAAG1ETR6LgALqRTDIGjIZ5KK4QnwwTCUMBaJAwZA2QQIbA2GkjGB5Ehk6C4GnoEiUZSgSCwRAoHas4BgADyEFwM0oyDBU1Ub00bOAhmMZgsOHwRBIpCceTAmFwAWQAA9kE5wUpXhpLMh0LoACp03BcoykVUwADUMCcAB0IIGrQE9Bt5QGICHtNkEAt7ENSTG4+MYABrN0eYwwDaadBwZlch36WnaIs84CUED9UjIujAM06HowKAwp0VyiY5utti8GDIiAIOhhcbIrswQfD8aYwyyQaONR2mB7Svq3TACa4ACSEHlPetC9QBdIIgAcsBT6QV5RuNxFQvSUeYGeL8ziDAGMB3jATLgnHsuRsLU1TCbRgDTYxDhbL9WXQFghC5EE2DQLhphUKEYSwZE0JUZEAnw+tGxAbs2w7TEcJUUN0GQ0hSGAZswjvAw5wo9D8EwmAwhgABaYxCQXBcrm4CjQw2eMmmbYpuFuASVAktDMX4xwrhgXAqy5edHFJHkRjYEwRC5QEEE0DNFmMXJOGwbkW1I3k0Iw2F+0IpsbLYdteR9ZdK0U65+nuIZkWRCSYGCppxzC4wQH7ScRzYMdmxi6cZO4IKfJtJUNHQagAkodBsH1Oh0FYRhem5Z1XVwVKD2VLKTVy-L+icYtmTLEqZJUYsXTLM5cFjeM6IYpjZDVX0wmbJppIeQliXSw0pRlOUwUeIA) where I only go through the data lines once total.

**Reply by bogdaaamn** [reply]
oh I love the single reducer, my 7 am brain could never 😂

**Reply by tkshill** [reply]
Love using the playground, and that fancy github api call. Happy to see you using TS :)

Also happy for comments, which I should add to mine. Great job!

**Reply by bogdaaamn** [reply]
I love how the comments are actually about the challenge (calling the elfs elfs)

**Comment by turingtestable** [comment]
This is my first pass in python ([part1](https://github.com/turingtestable/AdventOfCode2022/blob/master/challenge1/solution.py) [part2](https://github.com/turingtestable/AdventOfCode2022/blob/master/challenge1/solution2.py)) ,  i am going to muck through in another language at some point later today. @danieltott what language were you talking to me about over Thanksgiving that might be worth starting to learn this way.

**Reply by danieltott** [reply]
It was Elm! CC @tkshill 😁

**Reply by kierblk** [reply]
Elm has always been a language I wanted to learn.

**Reply by tkshill** [reply]
I will teach/support anyone here with Elm in a heartbeat. You only need to say the words

**Reply by turingtestable** [reply]
Yeah i was immediately intimidated by looking at setup and solutions from people for last year.

**Reply by doleraj** [reply]
Don't be. Some people go nuts with essentially building a framework specifically for AOC. I have a coworker who pre-wrote (or more likely copied from previous years) a bunch of helper methods. You don't need to do that. You can extract things out as you go, or you can switch languages every day, and it's still feasible to accomplish the month.

**Reply by turingtestable** [reply]
Yeah I am just used to backend languages for things like this where: I plop a file next to my solution, it sucks it up, and runs, so the setup of something running in a webserver by default just felt slightly more complicated. 

Especially since I have not touched elm before, so I have no idea about how to start it (ps if there is a good "how to start playing around with elm" site/doc whatever you want to throw my way I am happy to delve in). I have it pulled in with homebrew.

**Comment by kierblk** [comment]
I WANNA DO THIS!! 🌮

**Reply by danieltott** [reply]
<img src="https://media0.giphy.com/media/wi8Ez1mwRcKGI/giphy.gif"/>

**Comment by doleraj** [comment]
I used https://github.com/caderek/aocrunner this year to set me up since I am sadly bereft of spoons. I'll post my solution in a bit.

**Reply by danieltott** [reply]
Woah that is awesome!

**Reply by danieltott** [reply]
I thought I was all clever by pasting the input into a gist and loading it via fetch. This is really fancy

**Reply by tkshill** [reply]
I am trying to do a _very_ basic version of this for myself. But this is real neat!

**Reply by doleraj** [reply]
I'm hoping it gets me over the hump on actually being able to finish this year.

**Reply by doleraj** [reply]
My [TS solution](https://github.com/doleraj/adventofcode/blob/main/aoc2022/src/day01/index.ts). DRY is for suckers.

**Reply by tkshill** [reply]
I appreciate the dedication to context by prefixing your variable names with `elf`. Just so we know what the hell is going on.

**Reply by bogdaaamn** [reply]
WHAT this is an amazing tool

**Comment by tkshill** [comment]
Here's my [solution](https://github.com/tkshill/AdventOfCode2022/blob/main/AOC2022/Day01.fs). Going F# again? I think. That may change. 

Not particularly optimized or dry. Considering not refactoring part ones with updated info after seeing part two. So it stays true to how I thought of each.

**Reply by tkshill** [reply]
Every day the ECMAscript counsel refuses to add the pipe `|>` operator to js, my heart **weeps**

**Reply by bogdaaamn** [reply]
I save my efficiency and dry skills for future days, I can't burn out on the first day 😂

**Comment by turingtestable** [comment]
Added a ruby solution to mine too: https://github.com/turingtestable/AdventOfCode2022/blob/master/challenge1/solution.rb

**Reply by tkshill** [reply]
Ruby looks so much like python until it doesn't. Like, they each have some very specific optimizations. I feel like I'd mess up the syntax all the time.

**Reply by turingtestable** [reply]
Yeah i am definitely not perfect at writing idiomatic python or ruby, but i can get by and read both? It is sort of like being able to survive in a country where people speak Spanish or Portugese but occasionally flubbing one for the other lol.

**Reply by bogdaaamn** [reply]
🤔 never considered ruby for AoC, actually it's a good idea

**Reply by bogdaaamn** [reply]
don't mind if I do

**Comment by clandau** [comment]
I'm not letting myself go back and refactor 😄 https://github.com/clandau/advent-of-code/blob/main/2022/day-1/1.js

**Comment by sadiejay** [comment]
Just proud that I was able to figure it out, but I would like help breaking my mega functions into smaller functions lollol
https://github.com/sadiejay/aoc2022/tree/main/dec1

**Reply by turingtestable** [reply]
Have you ever used the [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) function? It might be a good way to make your loop into a more concise idea. Basically given some array it let's you run some function on every element of that array, and pass in the result of the previous execution of the function as an argument, which is really handy when you are trying to do something like "sum" values without needing to handle all the loop execution stuff.

That being said I think your solution is readable and not as "mega-sized" as you might think.

**Reply by sadiejay** [reply]
@turingtestable  Oh wow thank you!! I think `reduce` just became a new favorite haha

**Reply by bogdaaamn** [reply]
reduce everything!!!! learnt from @tkshill and never did it differently
